File: Queue/Queue.py
class Queue:
    def __init__(self):
        self.queue = []  # Инициализация пустого списка для очереди

    def enqueue(self, item):
        # Добавление элемента в конец очереди
        self.queue.append(item)

    def dequeue(self):
        # Удаление и возврат первого элемента очереди
        if not self.is_empty():
            return self.queue.pop(0)  # Извлечение элемента с начала
        else:
            print("Очередь пуста!")

    def peek(self):
        # Возврат первого элемента без удаления
        if not self.is_empty():
            return self.queue[0]
        else:
            print("Очередь пуста!")

    def is_empty(self):
        # Проверка, пуста ли очередь
        return len(self.queue) == 0

    def size(self):
        # Возврат количества элементов в очереди
        return len(self.queue)


# Пример использования очереди:
queue = Queue()
queue.enqueue(10)
queue.enqueue(20)
queue.enqueue(30)

print(queue.dequeue())  # Вывод: 10 (первый добавленный элемент)
print(queue.peek())  # Вывод: 20 (первый элемент)

File: Queue/queue.js
// Узел очереди
class QNode {
  constructor(value) {
    // Значение узла
    this.value = value;
    this.next = null;
  }
}

class Queue {
  constructor() {
    // Указатели на голову и хвост очереди
    this.head = null;
    this.tail = null;
    // Размер очереди
    this.size = 0;
  }

  // Добавление элемента в хвост очереди
  enqueue(value) {
    const node = new QNode(value);
    if (!this.tail) {
      this.head = node;
      this.tail = node;
    } else {
      this.tail.next = node;
      this.tail = node;
    }
    this.size++;
  }

  // Удаление элемента из головы очереди и возврат его
  dequeue() {
    if (!this.head) return undefined;
    const value = this.head.value;
    this.head = this.head.next;
    if (!this.head) this.tail = null;
    this.size--;
    return value;
  }
}

// Пример использования
const queue = new Queue();
queue.enqueue(10);
queue.enqueue(20);
console.log(queue.dequeue()); // 10
console.log(queue.dequeue()); // 20